### PR TITLE
Stop collecting source line information during a stacktrace for the E…

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
…xception.Source property.

Rough statistics when computed with and without the change. Note the callstack was only 2 deep so for deeper stacks this would be even more profound.

```
Scenario: get_Source - source file query [n = 4000] [values in ticks]
    Mean: 228
     Min: 63
     Max: 500755

Scenario: get_Source - NO source file query [n = 4000] [values in ticks]
    Mean: 52
     Min: 29
     Max: 10540
```

### CODE REVIEW

**Comment by stephentoub** [comment]
Is there any downside to this?  It seems like the information is ignored, so there's only upside in that we just don't do unnecessary work?

**Comment by AaronRobinsonMSFT** [comment]
@stephentoub Yep. That is what I was thinking. The cost to collect that addition data is rather high so when I stepped through this and verified it wasn't being used, it seems liked an obvious win.

**Comment by AaronRobinsonMSFT** [comment]
@jkotas Any concerns?

File: src/System.Private.CoreLib/src/System/Exception.cs [hunk 1]
```diff
@@ -337,7 +337,7 @@ public virtual string Source
             {
                 if (_source == null)
                 {
-                    StackTrace st = new StackTrace(this, true);
+                    StackTrace st = new StackTrace(this, false);
```
Comment (danmosemsft) [inline src/System.Private.CoreLib/src/System/Exception.cs]: Nit it is nice for the reader to name param in the case of a boolean literal ie
`new StackTrace(this, fNeedFileInfo: false)`

Some intersting Hungarian used there.

**Pull request merged**